nat/src/stateless/mod.rs

```diff
@@ -166,17 +210,16 @@ impl<Buf: PacketBufferMut> NetworkFunction<Buf> for StatelessNat {
         input: Input,
     ) -> impl Iterator<Item = Packet<Buf>> + 'a {
         input.filter_map(|mut packet| {
-            if !packet.is_done() {
+            if !packet.is_done() && packet.get_meta().nat {
```

Comment:
Nit: I'd maybe rename `.nat` to `.nat_required` or something like that, it would make the check clearer.

Reply:
After talking to Fredi, we can leave this, `nat_required` is long, we can decide how to name these flags later.